ballerina/googlesheets_client.bal

```diff
@@ -0,0 +1,520 @@
+        }
+        string|error textResponse = response.getTextPayload();
+        if (textResponse !is error) {
+            string[] responseRows = re `\n`.split(textResponse);
+            record {}[] rowTable = [];
+            if responseRows.length() == 0 {
+                return <Error>error("Error: the spreadsheet is not initialised correctly. ");
+            } else if responseRows.length() == 1 {
+                return rowTable.toStream();
+            }
+            string[] columnNames = re `,`.split(responseRows[0]);
+            foreach string rowString in responseRows.slice(1) {
+                int i = 0;
+                record {} rowArray = {};
+                string[] rowValues = re `,`.split(rowString);
+                foreach string rowValue in rowValues {
+                    string columnName = re `"`.replaceAll(columnNames[i], "");
+                    string value = re `"`.replaceAll(rowValue, "");
+                    string dataType = self.dataTypes.get(columnName).toString();
+                    if (dataType == "int") {
+                        (string|int|decimal|time:Date|time:TimeOfDay|time:Civil|time:Utc)|error typedValue = self.dataConverter(value, dataType);
+                        if typedValue is error {
+                            return <Error>error(typedValue.message());
+                        }
+                        rowArray[columnName] = <(string|int|decimal)>typedValue;
+                    } else if dataType == "time:Date" || dataType == "time:TimeOfDay" ||dataType == "time:Civil" || dataType == "time:Utc" {
+                        (string|int|decimal|time:Date|time:TimeOfDay|time:Civil|time:Utc)|error typedValue = self.dataConverter(value, dataType);
+                        if typedValue is error {
+                            return <Error>error(typedValue.message());
+                        } else if typedValue is time:Date {
+                            rowArray[columnName] = <time:Date>typedValue;
+                        } else if typedValue is time:TimeOfDay {
+                            rowArray[columnName] = <time:TimeOfDay>typedValue;
+                        } else if typedValue is time:Civil {
+                            rowArray[columnName] = <time:Civil>typedValue;
+                        } else if typedValue is time:Utc {
+                            rowArray[columnName] = <time:Utc>typedValue;
+                        }
+                    } else {
+                        (string|int|decimal|time:Date|time:TimeOfDay|time:Civil|time:Utc)|error typedValue = self.dataConverter(value, dataType);
+                        if typedValue is error {
+                            return <Error>error(typedValue.message());
+                        }
+                        rowArray[columnName] = <(string|int|decimal)>typedValue;
+                    }
+                    i = i + 1;
+                }
+                rowTable.push(rowArray);
```

Comment:
Can consider using [match statements](https://ballerina.io/learn/by-example/match-statement/) here. Code will be easier to read.

Reply:
Match statement is not a ideal solution here. please refer to [chat](https://discord.com/channels/957996897782616114/1107538772092014682)